pandas/_libs/internals.pyx

```diff
@@ -26,6 +30,10 @@ from pandas._libs.util cimport (
 )
 
 
+cdef extern from "Python.h":
```

Comment:
Surprised that this also makes a difference - Cython doesn't translate that `is None` call for a cdef object into the C-equivalent expression already?

Reply:
It does! But `PyWeakref_GetObject` returns a `PyObject*` which is not comparable to a `PyObject` (not-ponter) and `None` is the latter type. This is the reason for this quite ugly hack. If you know a better way I would really like to hear about it

Reply:
Did Cython give you a compilation error? I would expect `None` to be a pointer to a PyObject as well and for Cython to handle this within its native syntax.

Of course I am wrong a lot! So I might have the wrong mental model for what Cython is doing, but worth double checking this can't be expressed more naturally first

Reply:
FWIW I don't want to go down a rabbit hole on this - just double check real quick. If you spend anything more than 2 minutes trying to find an answer, let's just roll with what we've got

Reply:
No problem at all. Cython gives this error when using `None``
```
Error compiling Cython file:
------------------------------------------------------------
...
        # all these objects stay alive, e.g. df.items() for wide DataFrames
        # see GH#55245 and GH#55008
        if force or len(self.referenced_blocks) > self.clear_counter:
            self.referenced_blocks = [
                ref for ref in self.referenced_blocks
                if PyWeakref_GetObject(ref) is not None
                                            ^
------------------------------------------------------------

pandas/_libs/internals.pyx:914:44: Invalid types for 'is_not' (PyObject *, Python object)
```